togetherly/lib/views/widgets/custom_material_app.dart

```diff
@@ -0,0 +1,59 @@
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:togetherly/providers/app_providers.dart';
+import 'package:togetherly/providers/scaffold_provider.dart';
+import 'package:togetherly/themes.dart';
+import 'package:togetherly/views/widgets/scaffold_widgets.dart';
+import 'package:togetherly/views/screens/approval.dart';
+import 'package:togetherly/views/screens/child_home.dart';
+import 'package:togetherly/views/screens/parent_home.dart';
+import 'package:togetherly/views/screens/settings.dart';
+import 'package:togetherly/views/screens/store.dart';
+
+class CustomMaterialApp extends StatelessWidget {
+  const CustomMaterialApp({super.key, this.page});
+  final Widget? page;
+
+  Widget _getCurrentPage(int index, ScaffoldProvider provider) {
+    switch (index) {
```

Comment:
There aren't many options here. The only data I have to work with for the nav bar is the index of whichever nav item is clicked (0-n). That's how the widget keeps track of the current tab and how it updates; it's only through the index. I can't think of another way to accomplish this unless I were to build my own nav bar from scratch, which feels unnecessary. I know it feels rudimentary to manage the display with a switch/case of ints, but I think what I have right now is the best architecture that can be done since I'm managing the index with Provider and logically displaying the corresponding screen widget to its index. As for the title matching for child or parent, that logic will also change once we have implemented the other Providers and models.
In the future, when you have feedback like this, it would be more helpful to me and productive to our development if you offered an alternative implementation or additional ideas that can be considered rather than vague feedback that makes me guess at another solution.

Reply:
I suppose one thing I could do to make this simpler would be to make a list of widgets that contain each of the screens we are using, then, rather than using a switch/case statement to match each index, just simply return the screen from the list using the index to access it. This would follow the same principle, and would require that the index of the list is consistent with the index of the nav bar, but it would remove the need for a switch/case statement.

Reply:
My concern wasn't so much with it being a switch statement, but rather with it using numbers as a means of identifying pages. I think that makes perfect sense when managing state within a widget, but it becomes more problematic when you elevate it to app-wide state, because indices don't communicate a whole lot about what each value means. So you end up with a tendency to use magic numbers anywhere there is an interaction with that provider.

I apologize if my feedback seemed vague. I wasn't sure what the "standard" approach to this kind of thing was, and I didn't have time to research it myself. I was just unsure about the approach I was seeing, but you're right in that I should have substantiated my thoughts more. My intention was not to have you guess what I was thinking, as I didn't really have anything specific in mind.

I think we can stick with what you have. If anyone on the team comes across an approach that is meaningfully better, we can always refactor later.